### PR TITLE
feat(bundler): configurable parent Application name (--app-name)

### LINKED ISSUE #1011: feat(bundler): configurable parent Application name for multi-bundle clusters
## Summary

The `argocd-helm` and `argocd` deployers hardcode the parent Application name (`aicr-stack` and `nvidia-stack` respectively). When two bundles from different recipes (with non-overlapping components) are installed on the same cluster, the second bundle's parent Application silently overwrites the first, orphaning the first bundle's child Applications.

## Motivation / Context

Multiple CRDs targeting the same cluster would each install a bundle. Today this fails because both produce `name: aicr-stack` in the parent Application template.

## Current Behavior

**`argocd-helm` deployer** — `pkg/bundler/deployer/argocdhelm/argocdhelm.go:377`:
```yaml
metadata:
  name: aicr-stack      # hardcoded, no .Values override path
  namespace: argocd
```

**`argocd` deployer** — `pkg/bundler/deployer/argocd/templates/app-of-apps.yaml.tmpl:4`:
```yaml
metadata:
  name: nvidia-stack     # hardcoded
  namespace: argocd
```

Installing a second bundle overwrites the first parent Application. The first bundle's components become unmanaged.

## Expected Behavior

The parent Application name should be configurable so multiple bundles can coexist:

```bash
# Bundle A
aicr bundle -r gpu-recipe.yaml --deployer argocd-helm \
  --app-name gpu-runtime -o ./gpu-bundle

# Bundle B
aicr bundle -r ops-recipe.yaml --deployer argocd-helm \
  --app-name ops-runtime -o ./ops-bundle
```

## Proposed Solution

1. Add `--app-name` CLI flag to `aicr bundle`, defaulting to `aicr-stack` (argocd-helm) / `nvidia-stack` (argocd) for backward compatibility
2. In `argocd-helm`: expose via `.Values.appName` so it can also be overridden at `helm install` time
3. In `argocd`: template the name in `app-of-apps.yaml.tmpl`
4. Derive the template filename from the app name (e.g., `templates/<app-name>.yaml` instead of `templates/aicr-stack.yaml`)

## Additional Context

Audit of other hardcoded identifiers that affect multi-bundle:

| Identifier | Deployer | Workaround |
|-----------|----------|------------|
| Helm release name `aicr-bundle` | argocd-helm | Operator picks different name at `helm install` time |
| Chart name `aicr-bundle` in Chart.yaml | argocd-helm | Different OCI registry path at publish time |
| `argocd` namespace | both | Not a conflict — Argo CD manages many Applications in one namespace |
| Per-component Application names | both | Already unique (named by component) — safe for non-overlapping bundles |

The parent Application name is the **only blocker** with no operator workaround — it's a hardcoded string in the template, not reachable via `--set`.

### PR DESCRIPTION
**Pull request by mchmarny**
## Summary

Operators can now deploy multiple non-overlapping AICR bundles to the same Argo CD namespace by passing a distinct `--app-name` (or `app-name` query parameter) per bundle. The flag flows through CLI, API, and config-file surfaces and is honored by both the `argocd` and `argocd-helm` deployers.

## Motivation / Context

The `argocd` and `argocd-helm` deployers hardcoded the parent Argo Application name (`nvidia-stack` and `aicr-stack`). Two bundles installed into the same Argo CD namespace would each create a parent Application with the same name — the second overwrote the first, orphaning every child Application that the first had been managing. This blocks the multi-bundle topology intended for clusters that consume more than one recipe.

Fixes: #1011

## Type of Change

- [x] New feature (non-breaking change that adds functionality)

## Component(s) Affected

- [x] CLI (`cmd/aicr`, `pkg/cli`)
- [x] API server (`cmd/aicrd`, `pkg/api`, `pkg/server`)
- [x] Bundlers (`pkg/bundler`, `pkg/component/*`)
- [x] Docs/examples (`docs/`, `examples/`)

## Implementation Notes

**Plumbing:** `appName` flows through `pkg/bundler/config.Config` (private field, `AppName()` getter, `WithAppName` option) and `pkg/config/DeploymentSpec.AppName` (YAML/JSON `appName`). `BundleSpec.Resolve` validates the value as a DNS-1123 subdomain at the wire-to-typed boundary.

**Validation:** `config.ValidateAppName` (in `pkg/bundler/config`) returns `ErrCodeInvalidRequest` for non-DNS-1123 input. Empty is allowed and means "use the deployer's default." CLI/API both call it before constructing the bundler config.

**Deployer behavior:**
- `argocd-helm`: parent App template reads `{{ .Values.appName | default "aicr-stack" | quote }}`. When `--app-name` is supplied, the value is written into the bundle's root `values.yaml` so it is the chart default; `helm install --set appName=...` still wins. This matches the URL-portable contract that argocd-helm already follows for `repoURL`/`targetRevision`.
- `argocd`: the value is threaded into `AppOfAppsData.AppName` and `ReadmeData.AppName`, baked into the rendered `app-of-apps.yaml` and the README's `argocd app get/sync` examples at bundle time. This deployer materializes a static manifest, not a chart, so the choice cannot be deferred to apply time.

**Strictness:** `--app-name` / `app-name` is rejected on `--deployer helm` and `--deployer flux` (CLI returns `ErrCodeInvalidRequest`; API returns 400). Silently accepting it would mislead operators expecting their flag to take effect.

**Template filename:** kept fixed at `templates/aicr-stack.yaml` for argocd-helm even when `--app-name` is overridden. The rendered `metadata.name` is dynamic via `.Values.appName`; the filename is bundle-internal and stable across install-time `--set` overrides.

**Out of scope** (per #1011 thread):
- Flux template parent Kustomization name (separate doc-only fix)
- argocd-helm chart name `aicr-bundle` (already fixed in #1032)

## Testing

```bash
make qualify   # tests + lint + e2e + scan + license headers — all passed
```

New tests:
- `config.TestWithAppName`, `config.TestValidateAppName` — option round-trip + DNS-1123 validation table
- `argocdhelm.TestHelmTemplate_AppNameOverride` — live `helm template` covering default, bundle-time `AppName`, install-time `--set appName`, and bundle-time + install-time-override combo
- `argocd.TestGenerate_AppName` — default vs explicit `AppName` flows into both `app-of-apps.yaml` and README
- `cli.TestParseBundleCmdOptions_AppName` — flag parsing across argocd, argocd-helm, helm, flux deployers + invalid DNS rejection
- `bundler.TestBundleRequestAppNameParam` — API query-param accepted on argocd/argocd-helm, 400 on helm + invalid names
- Argocdhelm goldens regenerated for the new `.Values.appName` template line

## Risk Assessment

- [x] **Low** — Defaults preserve existing behavior verbatim (`aicr-stack` / `nvidia-stack` still in the rendered output for any caller that omits `--app-name`). The flag is gated by deployer kind so a typo on a helm/flux bundle fails loudly. Easy to revert.

**Rollout notes:** No migration required. Existing recipes and config files continue to bundle identically. Adopters with two bundles in one Argo CD namespace can now set distinct `--app-name` values.

## Checklist

- [x] Tests pass locally (`make test` with `-race`)
- [x] Linter passes (`make lint`)
- [x] I did not skip/disable tests to make CI green
- [x] I added/updated tests for new functionality
- [x] I updated docs if user-facing behavior changed (CLI reference, API reference, OpenAPI spec)
- [x] Changes follow existing patterns in the codebase
- [x] Commits are cryptographically signed (`git commit -S`)

### CODE REVIEW

File: docs/user/cli-reference.md [hunk 1]
```diff
@@ -1078,6 +1078,7 @@ aicr bundle [flags]
...
 | `--flux-oci-source-name` | | string | Name of the OCIRepository CR that Flux uses to pull the bundle (default: `aicr-bundle`). Used with `--deployer flux` and OCI output. Must match the OCIRepository deployed in the target cluster. See [Flux OCI Mode](#flux-oci-mode). |
 | `--flux-namespace` | | string | Kubernetes namespace where Flux CRs (HelmRelease, sources, ArtifactGenerator) are deployed (default: `flux-system`). Must match the namespace of the Flux installation in the target cluster. |
+| `--app-name` | | string | Parent Argo Application name (default: `aicr-stack` for `--deployer argocd-helm`, `nvidia-stack` for `--deployer argocd`). Must be a DNS-1123 subdomain. Required when deploying multiple non-overlapping AICR bundles to the same Argo CD namespace so the parent Applications do not collide. For `--deployer argocd-helm`, the value is the chart default and can still be overridden at install time via `helm install --set appName=...`. Rejected on `--deployer helm` and `--deployer flux`. |
```
[the remark this answers is not in the document]
Comment (mchmarny) [reply docs/user/cli-reference.md]: Addressed in d26f0691: updated the table cell to say "Rejected on other deployers (`helm`, `flux`, `helmfile`)" — explicit list with the parenthetical, matches API docs' broader phrasing while keeping the actual deployer names visible.

File: pkg/bundler/deployer/argocd/argocd.go [hunk 2]
```diff
@@ -216,6 +239,17 @@ func (g *Generator) VendorRecords() []localformat.VendorRecord {
...
+// appName returns the parent App-of-Apps `metadata.name` for this
+// Generator, applying the DefaultAppName fallback when the caller did
+// not set Generator.AppName. Centralized so the same value flows into
+// the rendered manifest, README examples, and any future audit fields.
+func (g *Generator) appName() string {
+	if g.AppName == "" {
+		return DefaultAppName
+	}
+	return g.AppName
+}
```
[the remark this answers is not in the document]
Comment (mchmarny) [reply pkg/bundler/deployer/argocd/argocd.go:252]: Addressed in d26f0691: `argocd.Generator.Generate` and `argocdhelm.Generator.Generate` now call `bundlercfg.ValidateAppName(g.AppName)` immediately after the `RecipeResult == nil` guard, so direct library callers can no longer ship a manifest whose name would only be rejected at apiserver admission. Empty still resolves to the deployer's default. New tests `TestGenerate_AppNameValidatedAtBoundary` in both deployer packages pin the behavior.

File: pkg/cli/bundle.go [hunk 3]
```diff
@@ -485,6 +504,17 @@ Package with explicit tag (overrides CLI version):
...
+				Name: "app-name",
+				Usage: "Parent Argo Application name (used by --deployer argocd and --deployer argocd-helm). " +
+					"Defaults: \"aicr-stack\" for argocd-helm, \"nvidia-stack\" for argocd. " +
+					"Override when deploying multiple non-overlapping AICR bundles to the same " +
+					"Argo CD namespace so the parent Applications do not collide. " +
+					"For --deployer argocd-helm, the value is the chart default and can be " +
+					"overridden at install time via `helm install --set appName=...`. " +
+					"Must be a DNS-1123 subdomain.",
+				Category: catDeployment,
+			},
```
[the remark this answers is not in the document]
Comment (mchmarny) [reply pkg/cli/bundle.go:517]: Addressed in d26f0691: added `"app-name"` to the `validateSingleValueFlags` list in `runBundleCmd` (alongside recipe, config, output, deployer, repo, storage-class). Also added a new `duplicate-bundle-app-name` step to `tests/chainsaw/cli/duplicate-flags` that exercises the rejection end-to-end via the real CLI binary (the pkg/cli unit-test helper goes through an Action override and bypasses runBundleCmd, so an integration-level check was the right home for the regression test).

**Review by yuanchen8911 — APPROVED** [review]

**Review by lalitadithya — APPROVED** [review]
